Feature #39363: Allow to switch between Any or All in tag filtering

diogotrodrigues opened the pull request:
#### As-Is:
When filtering DAGs using multiple tags in the Filter DAGs by tag field, an Any condition is applied.
This means DAGs with at least one of the specified tags are displayed.

#### To-Be:
Introduce the ability to choose between Any (default) and All operators for multi-tag filtering.
With the All operator, only DAGs containing all specified tags will be displayed.

#### What was done?:
We introduced a switch component to toggle between Any and All modes, which appears only when more than two Dag tags are selected, along with the necessary logic to support this functionality.
___
#### Here is a quick video demonstrating the feature!

https://github.com/user-attachments/assets/d9ad06bc-e074-47fb-b014-cc4805385c7e


<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---

diogotrodrigues committed:
Feature #39363: Allow to switch between Any or All in tag filtering

As-Is:
When filtering DAGs using multiple tags in the Filter DAGs by tag
field, an Any condition is applied.
This means DAGs with at least one of the specified tags are displayed.

To-Be:
Introduce the ability to choose between Any (default) and All
operators for multi-tag filtering.
With the All operator, only DAGs containing all specified tags will
be displayed.

What was done?:
We introduced a switch component to toggle between Any and All
modes, which appears only when more than two Dag tags are selected,
along with the necessary logic to support this functionality.

Co-authored-by: João Botelho <joao.v.botelho@tecnico.ulisboa.pt>

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/DagsList/DagsFilters.tsx:
```diff
@@ -231,6 +249,29 @@ export const DagsFilters = () => {
+        {selectedTags.length >= 2 && (
+          <HStack align="center" gap={1}>
+            <Text
+              color={tagFilterMode === "any" ? "blue.400" : "gray.500"}
```
Can we try to use semantic color tokens here? like fg.info and fg.muted? https://www.chakra-ui.com/docs/theming/colors#semantic-tokens

bbovenzi reviewed (approved):
lgtm just two nits

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/DagsList/DagsFilters.tsx:
```diff
@@ -34,6 +35,7 @@ import { useTableURLState } from "src/components/DataTable/useTableUrlState";
 import { QuickFilterButton } from "src/components/QuickFilterButton";
 import { StateBadge } from "src/components/StateBadge";
 import { Select } from "src/components/ui";
+import { Switch } from "src/components/ui/Switch.tsx";
```
```suggestion
import { Select, Switch } from "src/components/ui";
```

diogotrodrigues committed:
refactor: changed import and used semantic color tokens

Replaced hard-coded colors (blue.400 and gray.500) with
semantic tokens (fg.info and fg.muted) to improve theme consistency

Co-authored-by: João Botelho <joao.v.botelho@tecnico.ulisboa.pt>
